### PR TITLE
 [Translator] Add param & return type hints to all files in the Translator dir

### PR DESCRIPTION
**Pull request by tambait**
## Subject

This is a follow up of #6404 to add param & return type hints.

## Changelog

```markdown
### Added
- Added parameter and return type hints to all files in the Translator dir

### CODE REVIEW

**Comment by VincentLanglet** [comment]
I have trouble to understand why you created another branch/PR ? @tambait 
Can we close one ?

**Review by VincentLanglet — APPROVED** [review]

**Comment by tambait** [comment]
> I have trouble to understand why you created another branch/PR ? @tambait
> Can we close one ?
> 
> This one seems Ready to merge to me

Yes, please. I mistakenly create another branch/PR. Thanks.

**Comment by greg0ire** [comment]
Please squash the commits that fix cs into the commits that they fix

File: src/Translator/BCLabelTranslatorStrategy.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
  */
 final class BCLabelTranslatorStrategy implements LabelTranslatorStrategyInterface
 {
-    public function getLabel($label, $context = '', $type = '')
+    public function getLabel(string $label, string $context = '', string $type = ''): string
```
Comment (greg0ire) [inline src/Translator/BCLabelTranslatorStrategy.php:21]: This is a final class, which means the return type hint could be contributed to 3.x.

Please open a PR with the backwards-compatible part of your fixes on 3.x :pray:

File: src/Translator/Extractor/AdminExtractor.php [hunk 2]
```diff
@@ -76,7 +76,12 @@ public function __construct(Pool $adminPool, BreadcrumbsBuilderInterface $breadc
         $this->breadcrumbsBuilder = $breadcrumbsBuilder;
     }
 
-    public function extract($resource, MessageCatalogue $catalogue)
+    /**
+     * Extracts translation messages from files, a file or a directory to the catalogue.
+     *
+     * @param string|array $resource Files, a file or a directory
+     */
```
Comment (greg0ire) [inline src/Translator/Extractor/AdminExtractor.php:83]: This is backwards-compatible too

File: src/Translator/Extractor/AdminExtractor.php [hunk 3]
```diff
@@ -108,12 +113,19 @@ public function extract($resource, MessageCatalogue $catalogue)
         }
     }
 
+    /**
+     * NEXT_MAJOR: Add string type hint when support for Symfony 4 is dropped.
+     *
+     * Sets the prefix that should be used for new found messages.
+     *
+     * @param string $prefix The prefix
+     */
```
Comment (greg0ire) [inline src/Translator/Extractor/AdminExtractor.php:122]: And that too

File: src/Translator/BCLabelTranslatorStrategy.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Translator/BCLabelTranslatorStrategy.php:21]: The classes are not final on 3.x @greg0ire

**Review by VincentLanglet — APPROVED** [review]

**Review by wbloszyk — DISMISSED** [review]

**Comment by VincentLanglet** [comment]
Hi @tambait, this PR is ok but there is a conflict with master.

Can you rebase on master to fix them, then I'll merge it.

**Comment by tambait** [comment]
Hi @VincentLanglet thanks. I was about to work on this :)

**Review by wbloszyk — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks @tambait

**Pull request merged**